pkg/splunk/enterprise/configuration.go

```diff
@@ -1173,9 +1173,17 @@ func ValidateAppFrameworkSpec(ctx context.Context, appFramework *enterpriseApi.A
 		appContext.AppsStatusMaxConcurrentAppDownloads = splcommon.DefaultMaxConcurrentAppDownloads
 	}
 
+	appDownloadVolume := splcommon.AppDownloadVolume
+	t, err := os.Stat(splcommon.AppDownloadVolume)
+
 	// check whether the temporary volume to download apps is mounted or not on the operator pod
-	if _, err := os.Stat(splcommon.AppDownloadVolume); os.IsNotExist(err) {
-		scopedLog.Error(err, "Volume needs to be mounted on operator pod to download apps. Please mount it as a separate volume on operator pod.", "volume path", splcommon.AppDownloadVolume)
+	if os.IsNotExist(err) {
+		scopedLog.Error(err, "Volume needs to be mounted on operator pod to download apps. Please mount it as a separate volume on operator pod.", "volume path", appDownloadVolume)
+		return err
+	}
+
+	if err == nil && !t.IsDir() {
+		scopedLog.Error(err, "Volume needs to be mounted on operator pod to download apps. Please mount it as a separate volume on operator pod.", "volume path", appDownloadVolume)
```

Comment:
As discussed, plz backup and restore, as that helps other test cases to expect a proper value.